R/distribution_formulas.R

```diff
@@ -95,7 +95,8 @@ get_expected_name <- function(family, data_type) {
       family[["link"]] == "log" ~ "log_expected_index",
     data_type == "index" && grepl("lognormal|gaussian", family[["family"]]) &&
       family[["link"]] == "identity" ~ "expected_index",
-    grepl("comp", data_type) ~ "proportion_catch_numbers_at_age"
+    grepl("agecomp", data_type) ~ "proportion_catch_numbers_at_age",
+    grepl("lengthcomp", data_type) ~ "proportion_catch_numbers_at_length"
```

Comment:
Can we drop "comp" from this and elsewhere? There seems to be a disconnect between the data frame that has "age" and "length" and then the code which has "agecomp" and "lengthcomp" where I would like to see them be consistent. I know that the change from agecomp to ages was not originally intended for this pull request but I think it is a great place to get it done. If we need to wait until next month for the cleanup of M2 that is fine.

Reply:
I wasn't involved with the agecomp to age change decision. As stated above I don't think it's a good idea to force comp structure and population structure to be identical. This is obviously less important for age data than length where binning is much more common. I agree we should be consistent though so would prefer a separation between population_ages (or just ages if you really want) and comp_ages.

Reply:
we now have consistent use of comp everywhere.